server/src/main/java/org/opensearch/tasks/TaskInfo.java

```diff
@@ -150,6 +176,13 @@ public TaskInfo(StreamInput in) throws IOException {
         } else {
             resourceStats = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_2_8_0)) {
```

Comment:
Heads up—you might experience BWC test failures after this change. BWC tests launch a mixed cluster using your PR's commit and the latest commit in the 2.8 branch. Since this PR isn't backported to the 2.8 branch yet, it will cause serialization/deserialization issues. This is expected.

To minimize the impact for other contributors, we generally follow this multi-step process:
1. Create the original PR to the `main` branch with version checks against `V_3_0_0`. BWC tests shall pass.
2. Simultaneously create a backport PR to the `2.8` branch and a follow-up PR to the `main` branch with the desired version checks `V_2_8_0`.

Reply:
Thanks for this @ketanv3! I am keeping V_3_0_0 for now. Will change it to 2.8 later.